pkg/plugin/conntrack/_cprog/conntrack.c

```diff
@@ -123,10 +145,21 @@ static __always_inline bool _ct_create_new_tcp_connection(struct ct_v4_key key,
         return false;
     }
     new_value.eviction_time = now + CT_SYN_TIMEOUT;
-    new_value.flags_seen_tx_dir = flags;
+    new_value.flags_seen_tx_dir = p->flags;
     new_value.is_direction_unknown = false;
     new_value.traffic_direction = _ct_get_traffic_direction(observation_point);
+    new_value.conntrack_metadata.packets_forward_count = 1;
+    new_value.conntrack_metadata.bytes_forward_count = p->bytes;
+    // The initial SYN is captured. Set the traffic direction of the connection.
+    // This is important for the case where the SYN packet is not captured
+    // and the connection is created with unknown direction.
+    new_value.conntrack_metadata.traffic_direction = new_value.traffic_direction;
     bpf_map_update_elem(&retina_conntrack, &key, &new_value, BPF_ANY);
+    // Update packet
+    p->is_reply = false;
```

Comment:
Yes, this defaults to false if not set. It was added to be explicit and for code readability. Note, this statement was already existing, I moved it from the caller function for consistency, since it belongs with instructions of creating a new TCP connection.